src/js/components/SearchModal/SearchBar.js

```diff
@@ -31,65 +32,74 @@ export default class SearchBar extends Component {
-      const query = e.target.elements.searchInput.value;
-      if (!query) return;
+    this.addEvent(
+      'submit',
+      '#search-form',
+      throttle(this.handleSubmit.bind(this), 300)
+    );
+  }
 
-      rootStore.setState({ isLoading: true });
+  async handleSubmit(e) {
+    e.preventDefault();
+    const query = e.target.elements.searchInput.value;
+    if (!query) return;
 
-      const [error, data] = await getSearchAPI(query);
-      if (error) {
-        rootStore.setState({
-          isLoading: false,
-          status: { notFound: true, statusCode: error.statusCode },
-        });
+    rootStore.setState({ isLoading: true });
 
-        return;
-      }
+    const [error, data] = await getSearchAPI(query);
 
-      const { items, nextPageToken } = data;
-      if (!items.length) {
-        rootStore.setState({
-          status: { notFound: true, statusCode: 200 },
-        });
+    if (error) {
+      rootStore.setState({
+        isLoading: false,
+        status: { notFound: true, statusCode: error.statusCode },
+      });
 
-        return;
-      }
+      return;
+    }
 
+    const { items, nextPageToken } = data;
+    if (!items.length) {
       rootStore.setState({
-        searchOption: {
-          query,
-          pageToken: nextPageToken,
-        },
-        videos: addSavedToVideos(items),
-        status: { notFound: false, statusCode: 200 },
-        isLoading: false,
+        status: { notFound: true, statusCode: 200 },
       });
+
+      return;
+    }
+
+    rootStore.setState({
+      searchOption: {
+        query,
+        pageToken: nextPageToken,
+      },
+      videos: makeCardData(items, savedVideosStorage.load()),
+      status: { notFound: false, statusCode: 200 },
+      isLoading: false,
     });
   }
 }
 
-export function addSavedToVideos(rawVideos) {
-  const savedVideos = webStore.load();
+export function makeCardData(rawVideos, savedVideos) {
+  return addSavedToVideos(extractCardData(rawVideos), savedVideos);
+}
+
+function addSavedToVideos(videos, savedVideos) {
+  // TODO: 이중 for문 해결
+  return videos.map(video => {
+    const saved = savedVideos.some(
+      savedVideo => savedVideo.videoId === video.videoId
+    );
+    return { ...video, saved };
+  });
+}
 
-  return rawVideos.map(
-    ({
-      id: { videoId },
-      snippet: {
-        thumbnails: {
-          default: { url: thumbnailUrl },
-        },
-        title,
-        channelTitle,
-        publishTime,
-      },
-    }) => ({
-      videoId,
-      thumbnailUrl,
-      title,
-      channelTitle,
-      publishTime,
-      saved: savedVideos.includes(videoId),
-    })
-  );
+function extractCardData(rawVideos) {
+  return rawVideos.map(({ id, snippet }) => {
+    return {
+      videoId: id.videoId,
+      thumbnailUrl: snippet.thumbnails.default.url,
+      title: snippet.title,
+      channelTitle: snippet.channelTitle,
+      publishTime: snippet.publishTime,
+    };
+  });
```

Comment:
return에서 또 return으로 이어지는 구조이다 보니, 의미를 이해하기가 조금 어려워지는것 같아요. 아래와 같이 변수에 한번 담에서 return하는 방식은 어떨까요~?
```suggestion
  const videos = rawVideos.map(...)
  return videos
```

Reply:
아하! 가독성이 떨어지는 것 같네요. 수정해보았습니다.
수정 과정에서 사소한 질문 하나 드리고 싶습니다!

이런 메소드에서 변수에 저장하고 그 변수를 return 하는 경우가 있는데요. 
리팩터링에서 읽은 바로는, 마틴 파울러씨는 변수명을 result로 저장하고 사용하는 것을 선호한다고 하더군요.
`extractCardData` 같은 메소드의 경우, CardData를 추출해서 리턴하겠다는 의도가 이름에서 보이기 때문에 result와 같이 범용적인 이름으로 선언해도 무방하다고 생각이 들었습니다.

하지만 상세한 네이밍을 사용하라고 계속 이야기를 듣고 있어서, 이것이 적절할까 하는 고민이 듭니다.
혹시 준은 어떻게 생각하시는지 궁금해요!

[01cc2d3](https://github.com/woowacourse/javascript-youtube-classroom/pull/143/commits/01cc2d383c3718a4edc712c8afc908cf78a56486)

Reply:
좋은 질문인것 같아요!
저도 메서드 이름에서 의미가 명확하고, 메서드 내에 다른 변수들이 많이 생기는게 아니라면 `result`와 같은 추상적인 단어만 있어도, 그 의미가 충분히 추측이 가능하기 때문에 그렇게 작성할 것 같아요. 만약에 메서드 내에 로직이 많아져서 복잡해진다면 다른 변수명을 고민하겠지만 지금은 `result`로 충분할 것 같아보이네요 ㅎㅎ

Reply:
히히 준 섬세한 코멘트 감사합니다 ㅠ-ㅠ